tensorflow_serving/tools/docker/tests/dockerfile_devel_mkl_test.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+#
+# Tests if a Docker image built from Dockerfile.devel basically functions.
+#
+# It does this by loading up a half plus two toy model in the Docker image
+# and querying it, validating the response.
+#
+# The image passed to this test must be already available locally.
+#
+# Ex: $ bazel test :unittest_dockerfile_devel_mkl \
+#         --test_arg=tensorflow/serving:latest-devel-mkl \
+#         --test_output=streamed --verbose_failures
+
+declare -r PROJDIR=$(pwd)/tensorflow_serving
+source ${PROJDIR}/tools/docker/tests/docker_test_lib.sh || exit 1
+
+# Values to fill in for test
+# ------------------------------------------------------------------------------
+declare -r USE_NVIDIA_RUNTIME=false
+declare -r IS_DEVEL_IMAGE=true
+declare -r MODELDIR="${PROJDIR}/servables/tensorflow/testdata"
+declare -r MODELNAME="saved_model_half_plus_two_cpu"
```

Comment:
How do we know that mkl is working with this model? If mkl isn't compiled in, will this still pass?

Reply:
(another way of looking at this - we want to know when the mkl build is broken, either because it didn't build or because the optimizations are missing).

For the GPU builds, the model binds some ops to a GPU device, so it fails to work if you use the CPU image or if the GPU image is building without GPU support.

Reply:
I think checking that the model runs (the existing test), and then adding either option A or option B sounds like it would be sufficient.

Reply:
the additions would be part of the same test so it either all succeeds or fails.

Reply:
Sounds good. Karthik knows option A and B.  For B, you just need to run:

`ldd $(pip show tensorflow | grep Location | cut -d ":" - f2)/tensorflow/libtensorflow_framework.so | grep libmklml`

If it comes up empty, then there is no MKL. Otherwise, it returns something like:

```
(tf111_mkl_p36) [bduser@merlin-param01 ~]$ ldd $(pip show tensorflow | grep Location | cut -d ":" -f2)/tensorflow/libtensorflow_framework.so | grep libmklml
        libmklml_intel.so => /home/bduser/anaconda3/envs/tf111_mkl_p36/lib/python3.6/site-packages/tensorflow/../_solib_k8/_U@mkl_Ulinux_S_S_Cmkl_Ulibs_Ulinux___Uexternal_Smkl_Ulinux_Slib/libmklml_intel.so (0x00007ffba90ba000)
```

Reply:
yeah i think `A` is simpler option to go with. Gautham,  is there a way i could pass custom bash script which checks above to container. 

Enabling `MKLDNN_VERBOSE` will print lot of messages with mkl flags which is not required. And also we should enable that flag during build. So i stick to option A.

Reply:
I agree. A is cleaner and easier to implement.

Reply:
I would check for logs like we do to check to see when [the REST endpoint is up](https://github.com/tensorflow/serving/blob/master/tensorflow_serving/tools/docker/tests/docker_test_lib.sh#L67). Maybe add something similar (in this file) to look for those logs?

Reply:
Oh with option `A` we are not checking for anything in logs. 
We just execute this in `bash shell` and see if it returns anything as tony mentioned above.

Reply:
here i will add a condition like if docker_image_name contains `mkl` . then i will check for the output.
https://github.com/tensorflow/serving/blob/master/tensorflow_serving/tools/docker/tests/docker_test_lib.sh#L166

Reply:
You'll need to something like [this](https://github.com/tensorflow/serving/blob/master/tensorflow_serving/tools/docker/tests/docker_test_lib.sh#L155), except instead of running the model server, you're running your piped commands.

Then you can use something similar to my first link to check for the output from the docker container (those "logs") :)

Reply:
does that make senses to you?

Reply:
rather than check for mkl, add value called "IS_MKL_IMAGE=true" (see other config options there). That will work regardless of what the image is named.